agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +202,621 @@ def waitpid(pid):
+    """
+
+    envi_var = "PBENCH_TOOL_DATA_SINK_CFG"
+    def_port = def_wsgi_port
+    bad_port_ret_code = ReturnCode.BADWSGIPORT
+
+    def __init__(self, spec):
+        super().__init__()
+        if spec:
+            parts = spec.split(":", 1)
+            if len(parts) != 2:
+                raise self.Err(
+                    f"Bad Tool Data Sink specified, {spec!r}", ReturnCode.BADTDSARG
+                )
+            try:
+                self.port = int(parts[1])
+            except Exception as exc:
+                raise self.Err(
+                    f"Bad Tool Data Sink WSGI port specified, {spec!r}",
+                    self.bad_port_ret_code,
+                ) from exc
+            else:
+                self.host = parts[0]
+            # This instance is created by somebody else.
+            self._managed = False
+        else:
+            self.port = None
+            self.host = None
+            # This instance will be managed by us.
+            self._managed = True
+
+    def __str__(self):
+        return f"{self.host}:{self.port}"
+
+    def start(
+        self, exec_dir, full_hostname, tds_param_key, redis_server, redis_client, logger
+    ):
+        assert (
+            self._managed is True
+            and self.host is not None
+            and self.port is not None
+            and self.bind_host is not None
+            and self.bind_port is not None
+        ), f"Logic bomb!  Unexpected state: {self!r}"
+        try:
+            pid = os.fork()
+            if pid == 0:
+                # In the child!
+
+                # The main() of the Tool Data Sink module will not return here
+                # since it will daemonize itself and this child pid will be
+                # replaced by a new pid.
+                status = tds_main(
+                    [
+                        exec_dir / "tool-meister" / "pbench-tool-data-sink",
+                        redis_server.local_host,
+                        str(redis_server.port),
+                        tds_param_key,
+                        "yes",  # Request tool-data-sink daemonize itself
+                    ]
+                )
+                sys.exit(status)
+            else:
+                # In the parent!
+
+                # Wait for the child to finish daemonizing itself.
+                retcode = _waitpid(pid)
+                if retcode != 0:
+                    logger.error(
+                        "failed to create pbench data sink, daemonized; return code: %d",
+                        retcode,
+                    )
+        except Exception as exc:
+            raise self.Err(
+                "failed to create tool data sink, daemonized", ReturnCode.TDSFORKFAILED
+            ) from exc
+
+        # Wait for logging channel to be up and ready before we start the
+        # local and remote Tool Meisters.
+        timeout = time.time() + _TDS_STARTUP_TIMEOUT
```

Comment:
Maybe after 15-20 seconds of waiting add a message saying "Waiting for tool-data-sink..." or something similar, as I would probably assume something was broken or had crashed well before waiting a minute for start-up, since it usually takes a max of 2-3s. That also gives the user some time to start up their tds and save the run if they forgot to run it before using the `tool-meister-start` script.

Reply:
While this isn't a change from before, I'll see how to make that happen.

Reply:
Filed issue #2211 to track.